tests/integration-tests/tests/configure/test_pcluster_configure.py

```diff
@@ -161,3 +230,31 @@ def assert_config_contains_expected_values(
             validator.get("parameter_name")
         )
         assert_that(observed_value).is_equal_to(validator.get("expected_value"))
+
+
+@pytest.fixture(scope="module")
+def subnet_in_use1_az3(vpc_stacks):
+    # verify that use1_az3 does not have c5.xlarge or m6g.xlarge.
+    # If use1_az3 does not have the instance types,
+    # the fixture can be used to test the functionality of avoiding invalid subnet
+    ec2_client = boto3.client("ec2", region_name="us-east-1")
+    paginator = ec2_client.get_paginator("describe_instance_type_offerings")
+    page_iterator = paginator.paginate(
+        LocationType="availability-zone-id",
+        Filters=[
+            {"Name": "instance-type", "Values": ["c5.xlarge", "m6g.xlarge"]},
+            {"Name": "location", "Values": ["use1-az3"]},
+        ],
+    )
+    offerings = []
+    for page in page_iterator:
+        offerings.extend(page["InstanceTypeOfferings"])
+    if offerings:
+        logging.fatal(
+            "c5.xlarge or m6g.xlarge has been added to use1-az3, change the test code to test other rare instance type"
+        )
+    subnet_id = ec2_client.create_subnet(
+        AvailabilityZoneId="use1-az3", CidrBlock="192.168.16.0/20", VpcId=vpc_stacks["us-east-1"].cfn_outputs["VpcId"]
```

Comment:
Again, let's try to only pass in exactly what we need, in this case, `vpc_stacks["us-east-1"].cfn_outputs["VpcId"]`, but not the entire `vpc_stacks`
Please double check your code for this case